### ISSUE TITLE
Pumba is not an importable package

### ISSUE THREAD

**Issue by brice-morin** [alexei-led/pumba#60]
Hi,

I am writing a small Go library where I want to integrate with Pumba (and other Docker-related Go libs). It seems I am not allowed to import puma... `import ("github.com/gaia-adm/pumba")` does not work: `github.com/gaia-adm/pumba`

I have to say I am quite new to Go... But is there a way to make Pumba an importable package?

Thanks!

**Comment by alexei-led** [comment]
@brice-morin you are right. It was not planned to be "importable" since I did not think about use cases for embedding it into another program. 
This will require refactoring, and I have to estimate the effort.
PR is welcome too!

**Comment by brice-morin** [comment]
OK, I will have a look this week at making Pumba an importable package, and make a pull request if I manage to do something useful and clean

**Comment by brice-morin** [comment]
OK, I have made some experiments... Seems like Pumba's `action` and `container` packages already provide and export quite a clean API, which is then quite easy to import and use from another go program. I have been able to use Pumba's packet loss feature from my program, using my (rather dirty) fork of Pumba. Other features should be equally easy to integrate with.

I think the way to go is to split Pumba into two projects: Pumba API (everything except package `main`) and Pumba CLI (importing Pumba API and implementing package `main`)

Unfortunately, my fork is currently rather dirty, as I implemented some drastic measures, yes most likely not needed (it is just that I begin with Go), to get pumba to compile in my environment, which would make a PR at this point rather cumbersome to handle... But it at least shows the idea. Splitting Pumba in 2 will basically solve 99% of the problem for free.
